x/ibc-hooks/wasm_hook.go

```diff
@@ -202,3 +215,118 @@ func ValidateAndParseMemo(memo string, receiver string) (isWasmRouted bool, cont
 
 	return isWasmRouted, contractAddr, msgBytes, nil
 }
+
+func (h WasmHooks) SendPacketOverride(i ICS4Middleware, ctx sdk.Context, chanCap *capabilitytypes.Capability, packet ibcexported.PacketI) error {
+	concretePacket, ok := packet.(channeltypes.Packet)
+	if !ok {
+		return i.channel.SendPacket(ctx, chanCap, packet) // continue
+	}
+
+	isIcs20, data := isIcs20Packet(concretePacket)
+	if !isIcs20 {
+		return i.channel.SendPacket(ctx, chanCap, packet) // continue
+	}
+
+	isCallbackRouted, metadata := jsonStringHasKey(data.GetMemo(), types.IBCCallbackKey)
+	if !isCallbackRouted {
+		return i.channel.SendPacket(ctx, chanCap, packet) // continue
+	}
+
+	// We remove the callback metadata from the memo as it has already been processed.
+
+	// If the only available key in the memo is the callback, we should remove the memo
+	// from the data completely so the packet is sent without it.
+	// This way receiver chains that are on old versions of IBC will be able to process the packet
+
+	callbackRaw := metadata[types.IBCCallbackKey] // This will be used later.
+	delete(metadata, types.IBCCallbackKey)
+	bzMetadata, err := json.Marshal(metadata)
+	if err != nil {
+		return sdkerrors.Wrap(err, "Send packet with callback error")
+	}
+	stringMetadata := string(bzMetadata)
+	if stringMetadata == "{}" {
+		data.Memo = ""
+	} else {
+		data.Memo = stringMetadata
+	}
+	dataBytes, err := json.Marshal(data)
+	if err != nil {
+		return sdkerrors.Wrap(err, "Send packet with callback error")
+	}
+
+	packetWithoutMemo := channeltypes.Packet{
```

Comment:
```suggestion
	packetWithoutCallbackMemo := channeltypes.Packet{
```

Still has memo if multiple keys!

Reply:
renamed